app/src/main/java/com/kickstarter/ui/adapters/CommentsAdapter.kt

```diff
@@ -0,0 +1,45 @@
+package com.kickstarter.ui.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import com.kickstarter.R
+import com.kickstarter.databinding.EmptyCommentsLayoutBinding
+import com.kickstarter.databinding.ItemCommentCardBinding
+import com.kickstarter.models.Comment
+import com.kickstarter.ui.viewholders.CommentCardViewHolder
+import com.kickstarter.ui.viewholders.EmptyCommentsViewHolder
+import com.kickstarter.ui.viewholders.KSViewHolder
+import rx.Observable
+
+class CommentsAdapter(private val delegate: Delegate) : KSAdapter() {
+    interface Delegate : EmptyCommentsViewHolder.Delegate, CommentCardViewHolder.Delegate
+
+    @LayoutRes
+    override fun layout(sectionRow: SectionRow): Int {
+        return if (sectionRow.section() == 0) {
+            R.layout.item_comment_card
+        } else {
+            R.layout.empty_comments_layout
+        }
+    }
+
+    fun takeData(comments: List<Comment>) {
+        sections().clear()
+        addSection(Observable.from(comments).toList().toBlocking().single())
```

Comment:
Not fan of blocking observables, 
usually on the adapter for creating a section we do that on the init Block, take a look here https://github.com/kickstarter/android-oss/blob/a38f215357b1db5b276be80c6bd40b77bbb13642/app/src/main/java/com/kickstarter/ui/adapters/ActivityFeedAdapter.kt#L161 
or here https://github.com/kickstarter/android-oss/blob/a38f215357b1db5b276be80c6bd40b77bbb13642/app/src/main/java/com/kickstarter/ui/adapters/ProfileAdapter.kt#L23

Reply:
Fixed 
![Screen Shot 2021-05-24 at 3 32 26 PM](https://user-images.githubusercontent.com/1075310/119356148-6761fc80-bca6-11eb-9340-54762e77f82c.png)

Reply:
Remove part of handling empty state as app crash in case loggout user and it will be handle in different ticket 
![image](https://user-images.githubusercontent.com/1075310/119393598-74e0ac00-bcd1-11eb-9acc-ccb52e672807.png)